runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -3451,6 +3440,37 @@ JITaaSHelpers::getRemoteROMClassIfCached(ClientSessionData *clientSessionData, J
    return (it == clientSessionData->getROMClassMap().end()) ? NULL : it->second.romClass;
    }
 
+template <typename map, typename key>
+void JITaaSHelpers::purgeCache (std::vector<ClassUnloadedData> *unloadedClasses, map m, key ClassUnloadedData::*k)
+   {
+   ClassUnloadedData *data = unloadedClasses->data();
+   std::vector<ClassUnloadedData>::iterator it = unloadedClasses->begin();
+   while (it != unloadedClasses->end())
+      {
+      if (it->_cached)
+         {
+         m.erase((data->*k));
+         }
+      else
+         {
+         //If the class is not cached this is the place to iterate the cache(Map) for deleting the entry by value rather then key.
+         auto itClass = m.begin();
+         while (itClass != m.end())
+            {
+            if (itClass->second == data->_class)
+               {
+               m.erase(itClass);
+               break;
+               }
+            ++itClass;
+            }
+         }
+      //DO NOT remove the entry from the unloadedClasses as it will be needed to purge other caches.
+      ++it;
+      ++data;
+      }
+   }
+
```

Comment:
Since this is a templated method and it's implementation is only in this cpp file it can only be used in this file. Is that your intention? I'm OK with that, it seems like `purgeCache()` doesn't need to be called from anywhere else so there's no point in making it accessible to others, but if someone decides to call it from somewhere else they'll probably get a confusing linker error. Maybe you should add a comment to `purgeCache()` in the header explaining this.